src/codemagic/tools/_app_store_connect/arguments.py

```diff
@@ -357,6 +357,17 @@ def _apply_type(cls, non_typed_value: str) -> List[BetaBuildInfo]:
 
             return beta_build_infos
 
+    class DeviceUdidsArgument(cli.EnvironmentArgumentValue[List[str]]):
+        environment_variable_key = "DEVICE_UDIDS"
+        example_value = "00000000-000000000000001E"
+
+        @classmethod
+        def _apply_type(cls, non_typed_value: str) -> List[str]:
+            udids = non_typed_value.split()
+            if not udids:
```

Comment:
Depending on the way how `udids` list is constructed that check may not be sufficient. Consider input such as `'"" " "'`. Then from shell perspective the passed arguments are `['', ' ']` (empty string and just whitespace). We can be (quite) sure that neither are valid UDIDs.